### DISCUSSION TITLE
How do you set a blazor page/component internal (to avoid CA1515 warning)?

### DISCUSSION THREAD

**Discussion by Thieum** [dotnet/aspnetcore#58950] [Q&A]
Is it possible to seta blazor page / component as internal? marking the partial class in the code behind is not enough and will conflict with the default public visibility.

Example:

![image](https://github.com/user-attachments/assets/de0eeb67-55be-483d-b4cf-5b7e8b11a5c2)

```partial class AddDevice ``` will lead to a CA1515: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1515

```internal partial class AddDevice``` will lead to an error of inconsistent visibility : _Partial declarations of 'AddDevice' have conflicting accessibility modifiers_

**Comment by martincostello** [comment]
I've found it to be more trouble than it's worth and have just unconditionally turned it off in almost all my .NET 9 projects: https://github.com/dotnet/roslyn-analyzers/issues/7192
